### PR TITLE
Added signup, reset password, profile update, and deletion.

### PR DESCRIPTION
**Pull request by whatthestack**
http://127.0.0.1:8000/accounts/signup to start
Edit Profile / Logout in Header if user is logged in
Delete My Account under Edit Profile
Reset Password (not fully functional w/o email server)

@y3l2n, @Lucretiel - Please review.

### CODE REVIEW

**Review by Lucretiel — CHANGES_REQUESTED** [review]
Try to clean up your whitespace throughout these files; there are several instances of "no newline at the end of the file" or "mixed spaces and tabs".

File: backend/covid/forms.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            'first_name',
+            'last_name',
+            'email',
+            'password1',
+            'password2',
+            ]
+
+# Profile Form
+class ProfileForm(forms.ModelForm):
+
```
Comment (Lucretiel) [inline backend/covid/forms.py:24]: Delete this newlone

File: backend/covid/forms.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            'username',
+            'first_name',
+            'last_name',
+            'email',
+            ]
+
+class UserDeleteForm(forms.ModelForm):
+    class Meta:
+        model = User
+        fields = []   #Form has only submit button.  Empty "fields" list still necessary, though.
```
Comment (Lucretiel) [inline backend/covid/forms.py:37]: Newline at end of file

File: backend/covid/templates/covid/base.html [hunk 3]
```diff
@@ -22,6 +22,14 @@
...
 							Covid Community Resource Matching
 						</a>
+                        {%  if user.is_authenticated %}
+                        <a class="navbar-item" href="{% url 'userprofile' user.id%}">
+							Edit Profile
+						</a>
+                        <a class="navbar-item" href="{% url 'logout' %}">
+							Logout
+						</a>
+                        {% endif %}
```
Comment (Lucretiel) [inline backend/covid/templates/covid/base.html:32]: Mixed spaces and tabs

File: backend/covid_resource_matcher/urls.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 from django.urls import path, include
 
 urlpatterns = [
+    path("accounts/", include("covid.urls")),
```
Comment (Lucretiel) [inline backend/covid_resource_matcher/urls.py:20]: Don't apply *all* covid URLs to the `accounts/` route. Presumably this is for the user deletion endpoint; call out that url specifically (or even consider moving it to a separate urlpatterns)

File: backend/covid/templates/registration/password_reset_confirm.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{% extends "covid/base.html" %}
+
+{% block content %}
+    {% if validlink %}
+        <p>Please enter (and confirm) your new password.</p>
+        <form action="" method="post">
+        {% csrf_token %}
+            <table>
```
Comment (Lucretiel) [inline backend/covid/templates/registration/password_reset_confirm.html:8]: Prefer regular form fields over a table.

**Review by Lucretiel — CHANGES_REQUESTED** [review]

File: backend/covid_resource_matcher/urls.py (hunk 4 above)
Comment (SarahhhhFoster) [reply backend/covid_resource_matcher/urls.py:20]: @Lucretiel; I'm not savvy on much Django. What do you mean here? Would:
`path("accounts/delete/", include("covid.urls"))`
do the trick?

Comment (SarahhhhFoster) [reply backend/covid_resource_matcher/urls.py:20]: er... `path("deleteuser/", include("covid.urls")),`

Comment (Lucretiel) [reply backend/covid_resource_matcher/urls.py:20]: You're on the right track, but it'd be the opposite specialization. The problem with the code as written is that, for example, `/accounts/update_supplies` is a valid route, as is `/covid/deleteuser`. Your proposed solutions have a similar problem: `/accounts/delete/update_supplies` would be a valid route.

The solution depends on the final paths we want to have. The simplest solution would be to get rid of the `/accounts/` prefix and have all routes go through `/covid/` (eg, `/covid/deleteuser`, etc). In that case, we'd simply get rid of this line, and the account management routs would get picked up by `path("covid/", include("covid.urls"))`, below.

At a higher level, the way it works is that, for a given `path(...)`, call, all of the paths in the second argument are added as suffixes to the first argument. So, for example: `path("covid/", include("covid.urls"))` mounts all of the `covid.urls` paths onto `/covid/...`. The `covid.urls` paths are literally the paths in the `covid/urls.py` file, and specifically the paths described in `urlpatterns`. In turn, the urls in there may further `include(...)` other paths, or target specific view functions.

Feel free to email me if this still isn't clear; I'm happy to set up a video chat and give you a crash course in some of this stuff.

**Pull request merged**